seed_embeddings/OpenKE/analogy.py

```diff
@@ -27,6 +28,32 @@ def findVec(str1, src):
             return vec
 
 
+def findVecFromDict(str1, entity_dict):
+    """
+    Finds the vector for a given entity dictionary
+    """
+    if str1.upper() in entity_dict:
+        return np.array(entity_dict[str1.upper()])
+    else:
+        print(f"{str1} not found in entity_dict")
+    return None
+
+
+def genSimilarityTableFromDict(vec, entity_dict):
+    """
+    Generates cosine and Euclidean similarity tables based on the entity embeddings dictionary.
+    """
+    cosineDict = {}
+    euclDict = {}
+
+    for opcode, value in entity_dict.items():
+        value = np.array(value)
+        cosineDict[opcode] = spatial.distance.cosine(vec, value)
```

Comment:
Would using pairwise cosine_similarity speed up the computation? Please check. https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.cosine_similarity.html